docs/book/src/reference/testing/integration.md

```diff
@@ -77,4 +77,62 @@ func TestReconcile(t *testing.T) {
 
 The manager is started as part of the test itself (`StartTestManager` function).
 
-Both functions are located in `pkg/controller/parent/parent_controller_suite_test.go` file. The file also contains a `TestMain` function that allows you to specify CRD directory paths for the testing environment.
\ No newline at end of file
+Both functions are located in `pkg/controller/parent/parent_controller_suite_test.go` file. The file also contains a `TestMain` function that allows you to specify CRD directory paths for the testing environment.
+
+## Using envtest in your integration tests
+[`controller-runtime`]() offers `envtest` ([godoc](https://godoc.org/github.com/kubernetes-sigs/controller-runtime/pkg/envtest)), a package that helps write integration tests for your controllers by setting up and starting an api-server binary and an etcd binary.
+
+Using `envtest` in integration tests follows the general flow of:
+
+```
+import sigs.k8s.io/controller-runtime/pkg/envtest
+
+//specify testEnv configuration
+testEnv = &envtest.Environment{
+	CRDDirectoryPaths: []string{filepath.Join("..", "config", "crd", "bases")},
+}
+
+//start testEnv
+cfg, err = testEnv.Start()
+
+//write test logic
+
+//stop testEnv
+err = testEnv.Stop()
+```
+
+`kubebuilder` does the boilerplate setup and teardown of testEnv for you, in the ginkgo test suite that it generates under the `/controllers` directory.
+
+Logs from the test runs are prefixed `test-env`.
+
+### Configuring your test control plane
+You can use environment variables and/or flags to specify the `api-server` and `etcd` setup within your integration tests.
+
+#### Environment Variables
+
+| Variable name | Type | When to use |
+| --- | :--- | :--- |
+| `USE_EXISTING_CLUSTER` | boolean | Instead of setting up a local control plane, point to the control plane of an existing cluster. |
```

Comment:
```suggestion
| `USE_EXISTING_CLUSTER` | boolean | Instead of setting up a local control plane, point to the control plane of an existing cluster specified via the normal Kubernetes methods (in-cluster configuration, kubeconfig, etc). |
```